File: 3671-114-2130-maximum-twin-sum-of-a-linked-list/3671-114-2130-maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        ListNode slow = head,fast = head.next;

        while(fast!=null && fast.next!=null){
            slow = slow.next;
            fast = fast.next.next;
        }

        ListNode curr = null;
        while(slow!=null){
            ListNode temp = slow.next;
            slow.next = curr;
            curr = slow;
            slow = temp;
        }
        int ans = 0;
        while(head!=null && curr!=null){
            ans = Math.max(ans,head.val+curr.val);
            head = head.next;
            curr = curr.next;
        }
        return ans;
    }
}

File: 3671-114-2130-maximum-twin-sum-of-a-linked-list/3671-114-2130-maximum-twin-sum-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var pairSum = function(head) {
    let slow = head,fast = head.next
    while(fast && fast.next){
        slow = slow.next
        fast = fast.next.next
    }
    let ans = 0,node = null
    while(slow){
        let temp = slow.next
        slow.next = node
        node = slow
        slow = temp
    }

    while(head && node){
        ans = Math.max(ans,(node.val+head.val))
        node = node.next
        head = head.next
    }
    return ans
};